### ISSUE TITLE
Scenario Improvement: Vandalism affects rides and large scenery.

### ISSUE THREAD

**Issue by grahambilley** [OpenRCT2/OpenRCT2#11118]
**OS:** [Windows 10]
**Version:** [0.2.4]
**Commit/Build:** [d645338]

This is an enhancement request. I'm putting this here per request of Aaron van Geffen.

Allow angry guests the ability to "vandalize" rides and large scenery. For example:

1. Angry guests can cause a ride to break-down if enough angry guests are near it.
2. Angry guests can "destroy" large scenery causing it to disappear if enough angry guests are near it.

**Comment by James103** [comment]
For your first point, instead of angry guests directly causing a ride to break down, maybe make it so the ride ages faster if there are too many angry guests around it, causing it to lose value and reliability that much faster.

For your second point, maybe make it so instead of the scenery being destroyed, it is replaced by a ruined version that has a flag set to mark that it has been vandalized. When a guest looks at the vandalized scenery, it has the opposite effect on their happiness level compared to if the scenery was not vandalized. The cost to fix the scenery would be equal to the cost to build a new piece of this scenery minus some refund for salvageable parts of the vandalized scenery.

**Comment by grahambilley** [comment]
I like both points, especially the second one. I figured having it just disappear would be easier for the devs to incorporate into the code, but it would be a better feature the way you describe it.

Having angry guests directly destroy a ride would be a much more noticeable change to the game though, which is why I still like that option more than having it degrade faster. Imagine getting a pop-up message "Guests are angry and have vandalized Looping Roller Coaster 1 causing irreversible damage. You must tear it down and rebuild it."

**Comment by James103** [comment]
I updated my original response with the following idea: "If the reliability reaches zero because of [the angry guests], the ride crashes and closes down, you lose park rating, and some track pieces are deleted, forcing you to have to rebuild the entire coaster (or at least the destroyed sections) before you can reopen it."

**Comment by ocalhoun6** [comment]
Now hold on there ... that's a _lot_ of complicated stuff to talk about actually implementing.

Especially the scenery. You'd need new sprites for damaged scenery for _all_ the scenery in the game. Just a ton of new sprites.

And then you'd want options/cheats to be able to disable all of this.

It would be a _massive_ project to try and implement all of this.

**Comment by pizza2004** [comment]
Yeah, I like the other issues you've made, but I'm not particularly a fan of this one.  It feels too much like a deviation from what the game was meant to be about.  Rides are sacred and nothing should be modifying them without the player's consent first, in my opinion.  And with the scenery you'd either need to draw a ton of new sprites or you'd be removing pieces which could also upset people who don't remember exactly how they built something.

**Comment by grahambilley** [comment]
What if the scenery simply disappeared and a message popped up saying "guests are angry and are vandalizing your scenery". That was the original idea. No need to make new "broken" sprites for all the scenery in the game.

**Comment by grahambilley** [comment]
@pizza2004 I get that too. I'm more of a scenario player and less of a scenery person, so I'm just looking for things to make the game harder/more interesting. But I can totally see people who prefer to build parks into works of art viewing this idea as blasphemy.

**Comment by pizza2004** [comment]
It's more that Chris Sawyer built the game specifically around the idea of building roller coasters.  I care less about the scenery change and more about the idea that your rides might be partially destroyed.  In fact, a lot of people can't even stand the current vandalism as is implemented and I think they might have something like a cheat to turn that off.

If anything I'd prefer something like angry peeps stealing from shops or getting on rides without paying.  It would add some more variety to the things you need security guards instead of just being a more intense version of a game mechanic most players seem to dislike.

**Comment by grahambilley** [comment]
Angry people stealing merchandise and not paying for rides... now that is a good idea!

**Comment by James103** [comment]
@grahambilley What if angry guests could also steal money directly from ATMs? If they manage to steal from an ATM, a random amount of money (ranging from $1 up into the hundreds) would be taken out of your bank account and added to the guest's cash on hand.

**Comment by grahambilley** [comment]
@james103 I like that idea too. I think that we'll need to balance what the devs could implement easily with adding something to the game that the general player-base will enjoy. If people don't like the vandalism aspect of the game to begin with, then I doubt this enhancement will get legs behind it. Unfortunately I'm not a game developer and won't be able to implement any of these changes myself any time soon.

I find the scenarios very easy and a little repetitive, so most of my enhancements are to make the game harder and more interesting.

**Comment by James103** [comment]
Maybe this could be implemented as part of a plugin once the plugin system is merged, instead of directly onto the base game? I think it would be better suited as a plugin/s.
